app/src/androidTest/java/org/mozilla/focus/activity/AccessSettingsTest.java

```diff
@@ -86,4 +89,38 @@ public void AccessSettingsTest() {
         assertTrue(privacyHeading.exists());
         assertTrue(mozHeading.exists());
     }
+
+    @Test
+    public void BackRedirectionOnLearnMoreTest() {
+        UiObject searchHeading = TestHelper.mDevice.findObject(new UiSelector()
```

Comment:
I'm not super opposed to this way of testing the settings activity is visible when we press the back button, but this isn't ideal because it will be fragile wrt the settings activity (ie. it depends too much on the text in the activity). This might just be a case of YAGNI, but ideally all we should be checking is either:
a) a new session is not created when opening the "learn more" link
b) the InfoActitivy is created when opening the "learn more" link

Reply:
Not sure how to check the current activity is in a UI test. There is a deprecated method ``TestHelper.mDevice.getCurrentActivityName()``,  but it seems to output a weird string for InfoActivity.

Reply:
but I agree, this test is pretty fragile